source/screens/FormCaseScreen.tsx

```diff
@@ -54,30 +65,46 @@ const FormCaseScreen = ({
     initialCase?.forms?.[initialCase.currentFormId]?.answers || {};
 
   useEffect(() => {
-    if (caseId) {
-      const initCase = cases[caseId];
-
-      if (initCase !== undefined) {
-        // Beware, dragons! Since we pass by reference, it seems like the answers
-        // can be converted to object form already, thus we do this check.
-        if (Array.isArray(initCase?.forms?.[initCase.currentFormId]?.answers)) {
-          const answerArray =
-            initCase?.forms?.[initCase.currentFormId]?.answers || [];
-          const answersObject = convertAnswerArrayToObject(answerArray);
-          initCase.forms = {
-            ...initCase.forms,
-            [initCase.currentFormId]: {
-              ...initCase.forms[initCase.currentFormId],
-              answers: answersObject,
-            },
-          };
-        }
+    const setInitialCase = async () => {
+      if (caseId) {
+        const initCase = cases[caseId];
+
+        if (initCase !== undefined) {
+          // Beware, dragons! Since we pass by reference, it seems like the answers
+          // can be converted to object form already, thus we do this check.
+          if (
+            Array.isArray(initCase?.forms?.[initCase.currentFormId]?.answers)
+          ) {
+            const answerArray =
+              initCase?.forms?.[initCase.currentFormId]?.answers || [];
+            const answersObject = convertAnswerArrayToObject(answerArray);
+            initCase.forms = {
+              ...initCase.forms,
+              [initCase.currentFormId]: {
+                ...initCase.forms[initCase.currentFormId],
+                answers: answersObject,
+              },
+            };
+          }
+
+          const encryption =
+            initCase?.forms?.[initCase.currentFormId].encryption ?? {};
 
-        void getForm(initCase.currentFormId);
-        setCurrentFormId(initCase.currentFormId);
+          const pinCode =
+            ((await getPasswordForForm(
```

Comment:
`getPasswordForForm` is designed to throw when an "invalid" form is provided. Solo form (form without co-applicant) is considered invalid so it fails here. 

We could use `to` to catch this:

```typescript
const [, password] = await to(getPasswordForForm(...))
const passwordTouse password ?? ""
```